src/components/UserSelector.tsx

```diff
@@ -23,11 +48,20 @@ export const UserSelector: React.FC = () => {
 
       <div className="dropdown-menu" id="dropdown-menu" role="menu">
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+          {
+            users.map(user => (
+              <a
+                key={user.id}
+                href={`#user-#${user.id}`}
+                className={classNames('dropdown-item', {
+                  'is-active': user.id === selectedUser?.id,
+                })}
+                onClick={() => setSelectedUser(user)}
+              >
```

Comment:
Do not use anonymous function, create onUserClickHandler 
`const onUserClickHandler = (user) => () = > setSelectedUser(user)`
and use it `onClick={onUserClickHandler(user)}`